src/timetable/timetable.tsx

```diff
@@ -324,71 +327,76 @@ export const TimeTable = ({
-                ) : (
-                    <Loader $isEmptyContent={isEmptyContent}></Loader>
-                )}
-            </EmptyTableContainer>
-        );
-    }
-
-    return (
-        <TimeTableContainer
-            ref={tableContainerRef}
-            data-testid={testId}
-            {...otherProps}
-            $loading={loading}
-            $allRecordsLoaded={allRecordsLoaded || !onPage}
-        >
-            <RowColumnHeader
-                $isScrolledY={isScrolledY}
-                $isScrolledX={isScrolledX}
-            >
-                <TimeTableNavigator
+            <Container>
+                <TimeTableHeader
                     selectedDate={date}
                     loading={loading || loadMore}
                     tableContainerRef={tableContainerRef}
                     minDate={minDate}
                     maxDate={maxDate}
                     totalRecords={totalRecords}
-                    onLeftArrowClick={onPreviousDayClick}
-                    onRightArrowClick={onNextDayClick}
+                    onPreviousDayClick={onPreviousDayClick}
+                    onNextDayClick={onNextDayClick}
                     onRefresh={onRefresh}
+                    onCalendarDateSelect={onCalendarDateSelect}
                 />
-            </RowColumnHeader>
-            <RowHeaderColumn
-                $numOfRows={rowData.length}
-                $isScrolled={isScrolledX}
-                data-testid={"row-header-column-id"}
-            >
-                {renderRowHeaderColumn()}
-            </RowHeaderColumn>
-            <ColumnHeaderRow
-                $numOfColumns={hourlyIntervals.length}
-                $isScrolled={isScrolledY}
+                <EmptyTableContainer className="empty-container" {...otherProps}>
+                    <EmptyTableRowHeader>
+                    </EmptyTableRowHeader>
+                    {!loading ? (
+                        <NoResultsFound
+                            type="no-item-found"
+                            description={emptyContentMessage}
+                        />
+                    ) : (
+                        <Loader $isEmptyContent={isEmptyContent}></Loader>
+                    )}
+                </EmptyTableContainer>
+            </Container>
+        );
+    }
+
+    return (
+        <Container>
+            <TimeTableHeader
+                selectedDate={date}
+                loading={loading || loadMore}
+                tableContainerRef={tableContainerRef}
+                minDate={minDate}
+                maxDate={maxDate}
+                totalRecords={totalRecords}
+                onPreviousDayClick={onPreviousDayClick}
+                onNextDayClick={onNextDayClick}
+                onRefresh={onRefresh}
+                onCalendarDateSelect={onCalendarDateSelect}
+            />
+            <TimeTableContainer
+                ref={tableContainerRef}
+                data-testid={testId}
+                {...otherProps}
```

Comment:
you'll have to move the props `id`, `className`, `data-testid` to the `Container` as that's the main wrapper